src/models/step_arguments/data_table_spec.js

```diff
@@ -44,6 +44,235 @@ describe('DataTable', function () {
     })
   })
 
+  describe('typed hashes', function() {
+    it('rejects tables where not all rows have 3 columns', function() {
+      const dataTable = new DataTable({
+        rows: [
+          {
+            cells: [
+              {value: 'key1'},
+              {value: 'string'},
+              {value:  'value'}
+            ]
+          } , {
+            cells: [
+              {value: 'key2'},
+              {value: 'string'},
+              {value: 'value'}
+            ]
+          } , {
+            cells: [
+              {value: 'key3'},
+              {value: 'value'}
+            ]
+          }
+        ]
+      })
+      try {
+        dataTable.typedRowsHash()
+      } catch (err) {
+        expect(err.message).to.match(/typedRowsHash can only be used on a data table where all rows have exactly 3 columns/)
+        return
+      }
+      throw new Error('did not throw error for row that lacks columns')
+    })
+
+    it('rejects tables where not all types are recognized', function() {
+      const dataTable = new DataTable({
+        rows: [
+          {
+            cells: [
+              {value: 'key1'},
+              {value: 'string'},
+              {value:  'value'}
+            ]
+          } , {
+            cells: [
+              {value: 'key2'},
+              {value: 'OUPS'}, //<--- 1st No such type
+              {value: 'some value'}
+            ]
+          } , {
+            cells: [
+              {value: 'key3'},
+              {value: 'string'},
+              {value: 'value'}
+            ]
+          } , {
+            cells: [
+              {value: 'key3'},
+              {value: 'AYAYAYAYAY'}, //<--- 2nd No such type
+              {value: 'value'}
+            ]
+          }
+        ]
+      })
+
+      try {
+        dataTable.typedRowsHash()
+      } catch (err) {
+        expect(err.message).to.match(/typedRowsHash does not support type\(s\) in rows/)
+        const rows = err.message.split(/\n/).slice(1)
+        expect(rows.length).to.eql(2) //formatted row per rejected row
+        expect(rows[0].length).to.eql(rows[1].length) //formatted awsomely
+        return
+      }
+      throw new Error('did not throw error for unrecognized type')
+    })
+
+    it('reads value as string when no type is given', function() {
+      const dataTable = new DataTable({
+        rows: [
+          ['k5' , ''        , 'some value'],
+          ['j5' , ''        , 'other value']
+        ].map( (rawRow) => ({cells : rawRow.map((value) => ({value}))}) )
```

Comment:
This map fn appears to be repeated in a lot of places. Please create a helper at the top of this file. The helper could even just take in rows and build a data table to help dry up all these tests